synths/default-synths.scd

```diff
@@ -75,3 +75,197 @@ SynthDef(\gabor, { |out, sustain = 0.03, note = 0, amp = 1, pan = 0, width = 0.3
+	OffsetOut.ar(out, DirtPan.ar(sound, ~dirt.numChannels, 0.5));
+}).add;
+
+// now let's add some better duration control, and support for pan
+// Tidal's synth parameters magically show up as arguments to the SynthDef!
+// overall gain is handled elsewhere by SuperDirt, so we don't need it in the envelope
+SynthDef(\tutorial2, {|out, begin=0, end=1, pan, speed=1 |
+	var dur = (end-begin)/speed;
+	var env = EnvGen.ar(Env.linen(0.01, dur, 0.01, 1,-3), timeScale:dur, doneAction:2);
+	var sound = SinOsc.ar(440.0);
+	OffsetOut.ar(out, DirtPan.ar(sound, ~dirt.numChannels, pan, env));
+}).add;
+
+// it's also nice to control the pitch
+// using midicps means the "n" parameter will be used as MIDI note number
+// also, "accelerate" will cause the pitch to drift
+// in Tidal we'll need to say something like `sound "tutorial3:69"` to hear a reasonably high pitch
+SynthDef(\tutorial3, {|out, begin=0, end=1, pan, speed=1, accelerate, n |
+	var dur = (end-begin)/speed;
+	var env = EnvGen.ar(Env.linen(0.01, dur, 0.01, 1, -3), timeScale:dur, doneAction:2);
+	var sound = SinOsc.ar(n.midicps * Line.kr(1,1+accelerate));
+	OffsetOut.ar(out, DirtPan.ar(sound, ~dirt.numChannels, pan, env));
+}).add;
+
+// we can also make the envelope a more interesting percussive shape
+// note how speed affects the envelope
+SynthDef(\tutorial4, {|out, begin=0, end=1, pan, speed=1, accelerate, n |
+	var dur = (end-begin)/speed;
+	var env = EnvGen.ar(Env.perc(0.001/speed, 1/speed, 1, -4), timeScale:dur, doneAction:2);
+	var sound = SinOsc.ar(n.midicps * Line.kr(1,1+accelerate));
+	OffsetOut.ar(out, DirtPan.ar(sound, ~dirt.numChannels, pan, env));
+}).add;
+
+// finally, as an example of how to add your own parameters, let's say we want precise control over frequency
+// we'll make a new "f" parameter instead of "n" (don't use "freq", it's already used internally in SuperDirt)
+// to do this, we first need to this in Tidal: let (f, f_p) = pF "f" (Just 440)
+// then "f" is usable as an argument to the synthdef, and in Tidal you can try
+// d1 $ s "tutorial5/2" # f "[550,555]" # speed "1"
+SynthDef(\tutorial5, {|out, begin=0, end=1, pan, speed=1, accelerate, f |
+	var dur = (end-begin)/speed;
+	var env = EnvGen.ar(Env.perc(0.001/speed, 1/speed, 1, -4), timeScale:dur, doneAction:2);
+	var sound = SinOsc.ar(f * Line.kr(1,1+accelerate));
+	OffsetOut.ar(out, DirtPan.ar(sound, ~dirt.numChannels, pan, env));
+}).add;
+
+// physical modeling of a vibrating string, using a delay line (CombL) excited by an intial pulse (Impulse)
+// To make it a bit richer, I've combined two slightly detuned delay lines
+// "accelerate" is used for a pitch glide, and "speed" changes the envelope timescale
+SynthDef(\supermandolin, {|out, speed=1, begin=0, end=1, pan, accelerate, n, detune=0.2 |
+	var dur = (end-begin)/speed;
+	var env = EnvGen.ar(Env.linen(0.002, dur, 0.002, 1,-3), timeScale:dur, doneAction:2);
+	var sound = Decay.ar(Impulse.ar(0,0,0.1), 0.1*n/69) * WhiteNoise.ar;
```

Comment:
It's a decay time, but in the initial driving impulse rather than the sound itself.  And I just arrived at it rather arbitrarily after some experimentation.